Add cycle: create an infinite, looping range

Kijewski opened the pull request:
Rebased on #10.

Inspired by [itertools.cycle()][1].

[1]: https://docs.python.org/3.8/library/itertools.html#itertools.cycle

simonask commented:
I have taken the liberty of retargeting this PR as well to #10.

simonask commented on README.md:
```diff
@@ -96,6 +96,7 @@ following observations:
   storage, unless used as input for further algorithms.
 - `uniq()`: Reduce the output by consecutive equality with `std::unique()`. Does not allocate
   temporary storage, unless used as input for further algorithms.
+- `cycle()`: Create an infinite range repeating the inputs elements in a loop.
```
inputs -> input

simonask commented on include/rx/ranges.hpp:
```diff
@@ -1957,6 +1957,54 @@ struct reverse {
     }
 };
 
+/// Create an infinite range repeating the inputs elements in a loop.
```
inputs -> input

simonask reviewed (approved):
Good! Just spelling. :)

Kijewski committed:
Add cycle: create an infinite, looping range

Inspired by [itertools.cycle()][1].

[1]: https://docs.python.org/3.8/library/itertools.html#itertools.cycle

Kijewski commented:
Thank you for reviewing! Corrected.
